src/ui-kit/experimental/hierarchical/hierarchical/hierarchical.component.ts

```diff
@@ -77,6 +77,16 @@ export class SamHierarchicalComponent implements ControlValueAccessor {
   */
   @Input() selectedItemTemplate: TemplateRef<any>;
 
+  constructor(private cdr: ChangeDetectorRef) {
+
+  }
+  public singleMode: boolean = false;
+
+  ngAfterViewChecked() {
```

Comment:
@thomastighe  Same here -- if you use the `Checked` hooks, make sure to test performance.